api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -139,9 +139,11 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 							Obs replacement = os.saveObs(o, changeMessage);
 							//The logic in saveObs evicts the old obs instance, so we need to update
 							//the collection with the newly loaded and voided instance
-							toRemove.add(o);
-							toAdd.add(os.getObs(o.getId()));
-							toAdd.add(replacement);
+							if(!replacement.equals(o)) {
```

Comment:
The current logic is such that if you add a new member or remove an existing one, the parent is marked as dirty, my guess is that if you edit one of the child obs the parent is gonna come back as dirty because of [these calls](https://github.com/openmrs/openmrs-core/blob/2.0.0/api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java#L148-L153) and I'm not sure if this should be the case.